src/test/kotlin/com/example/passmanager/repositories/PassRepositoryImplTest.kt

```diff
@@ -11,207 +10,253 @@ import org.assertj.core.api.Assertions.assertThat
+            .flatMapMany { passRepository.getPassesPriceDistribution(it) }
 
         // THEN
-        assertThat(priceDistributions).hasSize(3).allMatch({ it.spentForPassType == BigDecimal.TEN })
+        priceDistributionFlux.collectList()
+            .test()
+            .assertNext { priceDistributions ->
+                assertThat(priceDistributions).hasSize(3).allMatch { it.spentForPassType == BigDecimal.TEN }
+            }
     }
 
     @Test
     fun `getting sum of purchased passes for pass owner should return correct sum`() {
         // GIVEN
-        val insertedPassOwner = mongoTemplate.insert(getOwnerWithUniqueFields())
-        val insertedPassOwnerId = insertedPassOwner.id
-        mongoTemplate.insertAll(passesToCreate.map { it.copy(passOwnerId = insertedPassOwnerId) })
+        // Cache the result to avoid re-triggering insert
+        val passOwnerMono = mongoTemplate.insert(getOwnerWithUniqueFields()).cache()
+        passOwnerMono.subscribe()
+
+        passOwnerMono
+            .map { owner -> passesToCreate.map { it.copy(passOwnerId = owner.id) } }
+            .flatMapMany { passes -> mongoTemplate.insertAll(passes) }
+            .subscribe()
+
         val afterDate = LocalDate.now()
 
         // WHEN
-        val sum = passRepository.sumPurchasedAtAfterDate(insertedPassOwnerId.toString(), afterDate)
+        val sum = passOwnerMono
+            .mapNotNull { it.id.toString() }
+            .flatMap { passRepository.sumPurchasedAtAfterDate(it, afterDate) }
 
         // THEN
-        assertThat(sum).isEqualTo(BigDecimal.valueOf(30))
+        sum.test()
+            .assertNext { assertThat(it).isEqualTo(BigDecimal.valueOf(30)) }
+            .verifyComplete()
     }
 
     @Test
     fun `getting sum of purchased passes for pass owner with no passes should return zero sum`() {
         // GIVEN
         val insertedPassOwner = mongoTemplate.insert(getOwnerWithUniqueFields())
-        val insertedPassOwnerId = insertedPassOwner.id
         val afterDate = LocalDate.now()
 
         // WHEN
-        val sum = passRepository.sumPurchasedAtAfterDate(insertedPassOwnerId.toString(), afterDate)
-
-        // THEN
-        assertThat(sum).isEqualTo(BigDecimal.ZERO)
-    }
-
-    @Test
-    fun `optimistic lock handling while save() should throw exception if version was changed by another thread`() {
-        // GIVEN
-        val createdPass = passRepository.insert(passToCreate)
-        val priceChangingList = listOf(BigDecimal.valueOf(20), BigDecimal.valueOf(30))
-        assertEquals(1, createdPass.version)
-
-        // WHEN
-        val tasks = priceChangingList.map {
-            Runnable { passRepository.save(createdPass.copy(purchasedFor = it)) }
-        }
-        val optimisticLocks = getOptimisticLocksAmount(tasks)
+        val sum = insertedPassOwner
+            .mapNotNull { it.id.toString() }
+            .flatMap { passRepository.sumPurchasedAtAfterDate(it, afterDate) }
```

Comment:
I think you can use insertedPassOwner.block() here, it's just tests :)

Reply:
in this particular test blocking inserting pass owner won`t change anything :)